cvat-ui/src/components/annotation-page/single-shape-workspace/single-shape-sidebar/single-shape-sidebar.tsx

```diff
@@ -0,0 +1,481 @@
+            minimalPoints = 1;
+        }
+
+        return minimalPoints;
+    };
+
+    if (action.type === ReducerActionType.SWITCH_AUTO_NEXT_FRAME) {
+        return {
+            ...state,
+            autoNextFrame: !state.autoNextFrame,
+        };
+    }
+
+    if (action.type === ReducerActionType.SWITCH_AUTOSAVE_ON_FINISH) {
+        return {
+            ...state,
+            saveOnFinish: !state.saveOnFinish,
+        };
+    }
+
+    if (action.type === ReducerActionType.SWITCH_COUNT_OF_POINTS_IS_PREDEFINED) {
+        return {
+            ...state,
+            pointsCountIsPredefined: !state.pointsCountIsPredefined,
+        };
+    }
+
+    if (action.type === ReducerActionType.SET_ACTIVE_LABEL) {
+        return {
+            ...state,
+            label: action.payload.label,
+            labelType: action.payload.labelType,
+            pointsCount: Math.max(state.pointsCount, getMinimalPoints(action.payload.labelType)),
+        };
+    }
+
+    if (action.type === ReducerActionType.SET_POINTS_COUNT) {
+        return {
+            ...state,
+            pointsCount: Math.max(action.payload.pointsCount, getMinimalPoints(state.labelType)),
+        };
+    }
+
+    return state;
+};
+
+function cancelCurrentCanvasOp(state: CombinedState): void {
+    const canvas = state.annotation.canvas.instance as Canvas;
+    if (canvas.mode() !== CanvasMode.IDLE) {
+        canvas.cancel();
+    }
+}
+
+function SingleShapeSidebar(): JSX.Element {
+    const appDispatch = useDispatch();
+    const store = useStore<CombinedState>();
+    const {
+        isCanvasReady,
+        jobInstance,
+        frame,
+        normalizedKeyMap,
+        keyMap,
+        defaultLabel,
+        defaultPointsCount,
+        navigationType,
+    } = useSelector((_state: CombinedState) => ({
+        isCanvasReady: _state.annotation.canvas.ready,
+        jobInstance: _state.annotation.job.instance as Job,
+        frame: _state.annotation.player.frame.number,
+        keyMap: _state.shortcuts.keyMap,
+        normalizedKeyMap: _state.shortcuts.normalizedKeyMap,
+        defaultLabel: _state.annotation.job.queryParameters.defaultLabel,
+        defaultPointsCount: _state.annotation.job.queryParameters.defaultPointsCount,
+        navigationType: _state.annotation.player.navigationType,
+    }), shallowEqual);
+
+    const [state, dispatch] = useReducer(reducer, {
+        autoNextFrame: true,
+        saveOnFinish: true,
+        pointsCountIsPredefined: true,
+        pointsCount: defaultPointsCount || 1,
+        labels: jobInstance.labels.filter((label) => label.type !== LabelType.TAG && label.type !== LabelType.SKELETON),
+        label: null,
+        labelType: jobInstance.labels[0]?.type || LabelType.ANY,
+    });
+
+    const savingRef = useRef(false);
+    const nextFrame = useCallback((): void => {
+        (frame < jobInstance.stopFrame ? jobInstance.annotations.search(frame + 1, jobInstance.stopFrame, {
```

Comment:
May we rewrite this with ordinary `if`? Ternary operator with length of 20 lines is quite hard to read

Reply:
Done